crypto/commitments/damgard-fujisaki.go

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2017 XLAB d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package commitments
+
+import (
+	"github.com/xlab-si/emmy/crypto/common"
+	"math/big"
+	"errors"
+	"github.com/xlab-si/emmy/crypto/groups"
+)
+
+func DamgardFujisaki() (bool, error){
+	safePrimeBitLength := 1024
+	receiver, err := NewDamgardFujisakiReceiver(safePrimeBitLength)
+	if err != nil {
+		return false, err
+	}
+	committer := NewDamgardFujisakiCommitter(receiver.QRSpecialRSA.N, receiver.H, receiver.G)
+
+	a := common.GetRandomInt(receiver.QRSpecialRSA.N)
+	c, err := committer.GetCommitMsg(a)
+	if err != nil {
+		return false, err
+	}
+
+	receiver.SetCommitment(c)
+	committedVal, r := committer.GetDecommitMsg()
+	success := receiver.CheckDecommitment(r, committedVal)
+
+	return success, nil
+}
+
+type DamgardFujisakiCommitter struct {
+	QRSpecialRSA 		*groups.QRSpecialRSA
+	H               *big.Int
+	G               *big.Int // G = H^alpha % N
+	committedValue  *big.Int
+	r               *big.Int
+}
+
+func NewDamgardFujisakiCommitter(n, h, g *big.Int) *DamgardFujisakiCommitter {
+	group := groups.NewQRSpecialRSAPublic(n)
+	return &DamgardFujisakiCommitter{
+		QRSpecialRSA: group,
+		H: h,
+		G: g,
+	}
+}
+
+func (committer *DamgardFujisakiCommitter) GetCommitMsg(a *big.Int) (*big.Int, error) {
+	group := committer.QRSpecialRSA
+	if a.Cmp(group.N) != -1 {
+		err := errors.New("the committed value needs to be < N") // TODO: to be checked
```

Comment:
I don't understand the TODO. Will you implement it in scope of this PR?
EDIT: okay, according to our offline discussion, I bet this is indeed coming in one of the follow-up PRs.

Reply:
It needs to be checked which values can be committed to (will do it in next PR).